presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -146,10 +152,35 @@ private SystemAccessControl create(String configFileName)
                     Optional.of(Pattern.compile(".*")),
                     Optional.of(Pattern.compile("system"))));
 
-            return new FileBasedSystemAccessControl(catalogRulesBuilder.build(), rules.getPrincipalUserMatchRules());
+            return new FileBasedSystemAccessControl(catalogRulesBuilder.build(), rules.getImpersonationRules(), rules.getPrincipalUserMatchRules());
         }
     }
 
+    @Override
+    public void canImpersonateUser(SystemSecurityContext context, String userName)
+    {
+        if (!impersonationRules.isPresent()) {
+            // if there are principal user match rule, we assume that impersonation checks are
+            // handled there; otherwise, impersonation must be manually configured
+            if (!principalUserMatchRules.isPresent()) {
```

Comment:
is it legal situation when `! impersonationRules.isPresent() && !principalUserMatchRules.isPresent()`?
if so, it could be rejected in the ctor

Reply:
If I'm reading correctly, you are asking "is it legal to have no impersonation rules and no principal rules"?  Absolutely.  It means that you are relying on user extraction to to determine which principals are allowed, and you want no impersonation.